### ISSUE TITLE
ssh_exchange_identification: Connection closed by remote host

### ISSUE THREAD

**Issue by raqqa** [nodester/nodester#346]
Worked just fine yesterday, but now I get this problem.

$ git push nodester master
ssh_exchange_identification: Connection closed by remote host
fatal: The remote end hung up unexpectedly

So I tried setting my key again,

$ nodester user setkey
nodester info sending sshkey: <pathtomykey>\id_rsa.pub
nodester info ssh key successfully sent

But still doesn't work.

I don't know if related, but I also get

$ nodester npm list <myappname>
nodester info list npm packages for app: <myappname>
nodester /usr/bin/n: line 38: /dev/null: Permission denied
nodester /usr/bin/n: line 41: /dev/null: Permission denied

EDIT;
I'm stupid, didn't see the previous issue on the same subject.

**Comment by nodester** [comment]
Hi Mikael,

We are currently running extremely low on disk space which is causing this
issue.  We are in the process of adding more resources to the platform as
we speak.

Thanks,
Chris

On Mon, Mar 19, 2012 at 11:18 AM, Mikael Emilsson <
reply@reply.github.com

**Comment by raqqa** [comment]
Yeah, saw the other issue.

Is this what's causing the npm list issue as well?

**Comment by nodester** [comment]
yes :(

On Mon, Mar 19, 2012 at 12:39 PM, Mikael Emilsson <
reply@reply.github.com

**Comment by alejandro** [comment]
@raqqa can you try again please?  So we can close this issue :)

**Comment by raqqa** [comment]
Works fine now, thanks guys!

**Comment by Domiii** [comment]
I am having this issue now, too.

**Comment by alejandro** [comment]
Hey @Domiii what's your problem? THe ssh stuff? Try to restart your rsa keys.

**Comment by Domiii** [comment]
Ok I just tried it again 5 minutes later and it just works. Thanks for the quick response anyway :)

**Comment by alejandro** [comment]
No problem :)! Sometimes is because the internet connection.

Hack the planet \m/

**Comment by ghost** [comment]
it seems this same issue is back

**Comment by ctm** [comment]
It's happening a bunch to us (Jackrabbit Systems), too.  It is intermittent, but it interferes with doing deploys.  We may have six servers all pulling from github successfully with one failing.  I just had it happen minutes ago.

**Comment by bubba-h57** [comment]
Indeed. We have been seeing it for the past day or so. I initially thought it was our server, but I have now attempted it on enough different machines to be fairly certain it is not. It is always intermittent as well, a few additional manual attempts will eventually yield a good pull or push.

**Comment by alejandro** [comment]
I was holding my voice here because I actually don't know what's the origin of this problem. I'd like to believe that it's caused because our instance.  `ssh` stuff is a OS based system. Well if you the process fails on a push there is nothing to do with "nodester" itself. So we don't have too much to do there. 

Checking the logs and everything looks fine. Probably is a thing with aws. 

What you guys can try is to reset your rsa keys, maybe that help.

@ctm Jackrabbit is hosted on nodester? Or you have installed the nodester server?.

Cheers.

**Comment by kensmith** [comment]
This is happening again today. Is it a low disk space issue again? Is there an ETA to resolution?

**Comment by wapcaplet** [comment]
+1, I'm seeing this today when trying to push or pull in a repository that was working fine before:

```
$ git pull origin develop
ssh_exchange_identification: Connection closed by remote host
fatal: The remote end hung up unexpectedly
$ git push origin develop
ssh_exchange_identification: Connection closed by remote host
fatal: The remote end hung up unexpectedly
```

Edit: It's working OK now.

**Comment by kcantwel** [comment]
+1:  Seeing this as well. 

% git push origin master
ssh_exchange_identification: Connection closed by remote host
fatal: The remote end hung up unexpectedly

**Comment by brasten** [comment]
Same here, started this morning.

**Comment by alejandro** [comment]
cc @chrismatthieu

**Comment by chrismatthieu** [comment]
AppFog is supporting the Nodester platform now.  Please send emails to support@appfog.com.  Thanks.

**Comment by ghost** [comment]
I'm also getting this message when doing pull. Trying from multiple computers.
ssh_exchange_identification: Connection closed by remote host

**Comment by ghost** [comment]
...and now it suddenly works to pull again. What is happening??! Can you please try to find the underlying cause for this?

**Comment by mikelyons** [comment]
Had this issue, kept trying, worked on the 8th try ...

Trying to `git pull --rebase`

**Comment by ihoneymon** [comment]
@mikelyons Thanks. I solved problem. ^^

**Comment by alexpogrebnyak** [comment]
@ihoneymon Hi, what did you do to solve problem? I have the same:(
